### ISSUE TITLE
Don't set `linkerd.io/proxy-version: ""` on injected workloads

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#9311]
Injected workloads appear to always have an annotation set `linkerd.io/proxy-version: ""`. If the value is empty, the annotation should be omitted. OR we should always set it with a valid value.

**Comment by jeremychase** [comment]
I was unable to reproduce using `edge-22.8.3` or a locally built `dev-` version of linkerd.

**Comment by olix0r** [comment]
Hmmm, this is what I see:

```
; k get deploy -o yaml -n emojivoto | grep -c proxy-version
0
:; k get po -o yaml -n emojivoto | grep proxy-version                  
      linkerd.io/proxy-version: ""
      linkerd.io/proxy-version: ""
      linkerd.io/proxy-version: ""
      linkerd.io/proxy-version: ""
:; k get deploy -o yaml -n emojivoto | grep -c proxy-version
0
:; k get ns -o yaml emojivoto | grep -c proxy-version   
0
:; linkerd version
Client version: edge-22.8.3
Server version: stable-2.12.0
```

**Comment by adleong** [comment]
```
> linkerd inject https://run.linkerd.io/emojivoto.yml | k apply -f -
[...]
> k get po -o yaml -n emojivoto | grep proxy-version
      linkerd.io/proxy-version: stable-2.12.0
      linkerd.io/proxy-version: stable-2.12.0
      linkerd.io/proxy-version: stable-2.12.0
      linkerd.io/proxy-version: stable-2.12.0
> k get cm -n linkerd linkerd-config -o yaml | yq .data.values | yq .proxy.image
name: cr.l5d.io/linkerd/proxy
pullPolicy: ""
version: stable-2.12.0
```

looks like `.proxy.image.version` is unset in your linkerd config for some reason...

**Comment by olix0r** [comment]
That's the default in helm:

https://github.com/linkerd/linkerd2/blob/b7387820c315bd769380267235cfaa69aeef269d/charts/linkerd-control-plane/values.yaml#L101-L118

**Comment by adleong** [comment]
Ah, I think I see what's happening.  This value is populated programmatically here: https://github.com/linkerd/linkerd2/blob/main/pkg/charts/linkerd2/values.go#L247

This means that it will be set if Linkerd is installed with the CLI but not if Linkerd was installed with Helm.

Every place that this value is used in the templates should already be applying the correct defaulting behavior so it's fine that this isn't set.  In fact, we should probably STOP setting this in the CLI install flow in order to have more consistency between the CLI and Helm flows.  

However, it looks like we're missing the defaulting logic in the inject codepath which sets the proxy-version annotation.  This codepath should fall back to using the Linkerd version if the proxy version isn't set.

**Comment by adleong** [comment]
Fixed by #9382
